src/rulesets/LinearAlgebra/structured.jl

```diff
@@ -63,18 +63,84 @@ function rrule(::typeof(*), D::Diagonal{<:Real}, V::AbstractVector{<:Real})
 end
 
 #####
-##### `Symmetric`
+##### `Symmetric`/`Hermitian`
 #####
 
-function rrule(::Type{<:Symmetric}, A::AbstractMatrix)
-    function Symmetric_pullback(ȳ)
-        return (NO_FIELDS, @thunk(_symmetric_back(ȳ)))
+function frule((_, ΔA, _), T::Type{<:LinearAlgebra.HermOrSym}, A::AbstractMatrix, uplo)
+    return T(A, uplo), T(ΔA, uplo)
+end
+
+function rrule(T::Type{<:LinearAlgebra.HermOrSym}, A::AbstractMatrix, uplo)
+    Ω = T(A, uplo)
+    function HermOrSym_pullback(ΔΩ)
+        return (NO_FIELDS, @thunk(_symherm_back(T, ΔΩ, Ω.uplo)), DoesNotExist())
```

Comment:
Do we need a thunk here? The cotangent w.r.t. A is the only meaningful, so I would imagine that it would always get used somewhere, but my intuition for this isn't great, perhaps yours is better.

Reply:
Probably not? I had thought that the convention was to always thunk in reverse unless 1) there's only one cotangent (not counting the `NO_FIELDS`) or 2) sometimes for scalar functions (e.g. those made with `@scalar_rule`). I personally think modifying (1) to be "there's only one cotangent that a user could reasonably want" makes perfect sense.

Reply:
@oxinabox do you agree? If so I'll open a PR to the docs to update guidance.